# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1377722103 14400
#      Wed Aug 28 16:35:03 2013 -0400
# Node ID 6470a113eae21563bd626e260257978f84cd0e4b
# Parent  091e4df179de26eebea846d52914af051141fae7
correctly handle deprecated \ continuation markers

* lex.ll (\\{S}*{NL} | \\{S}*{CCHAR}.*{NL}):
Call curr_lexer->handle_continuation.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -982,16 +982,18 @@ ANY_INCLUDING_NL (.|{NL})
     std::string nm = curr_lexer->fcn_file_full_name;
 
     if (nm.empty ())
       warning_with_id ("Octave:deprecated-syntax", "%s", msg);
     else
       warning_with_id ("Octave:deprecated-syntax",
                        "%s; near line %d of file '%s'", msg,
                        curr_lexer->input_line_number, nm.c_str ());
+
+    curr_lexer->handle_continuation ();
   }
 
 %{
 // End of file.
 %}
 
 <<EOF>> {
    return curr_lexer->handle_end_of_input ();
